app/models/work.rb

```diff
@@ -487,16 +508,6 @@ def past_snapshots
     UploadSnapshot.where(work: self)
   end
 
-  # Remove the snapshots for S3 file resources which have been deleted
-  def valid_snapshots
-    s3_resource_urls = s3_files.map(&:url)
-    s3_resource_filenames = s3_files.map(&:filename)
-
-    past_snapshots.select do |snapshot|
-      s3_resource_urls.include?(snapshot.url) || s3_resource_filenames.include?(snapshot.filename)
-    end
-  end
-
```

Comment:
We don't use this method. I noticed that is extra while troubleshooting this issue since we initially thought the problem was with the "snapshots code"